packages/sitecore-jss/src/i18n/graphql.dictionary-service.ts

```diff
@@ -0,0 +1,225 @@
+   * @default Search query
+   * query DictionarySearch(
+   * $rootItemId: String!,
+   * $language: String!,
+   * $dictionaryEntryTemplateId: String!,
+   * $pageSize: Int = 10,
+   * $after: String
+   * ) {
+   * search(
+   * where: {
+   * AND:[
+   * { name: "_path",      value: $rootItemId },
+   * { name: "_templates", value: $dictionaryEntryTemplateId },
+   * { name: "_language",  value: $language }
+   * ]
+   * }
+   * first: $pageSize
+   * after: $after
+   * orderBy: { name: "Title", direction: ASC }
+   * ) {
+   * total
+   * pageInfo {
+   * endCursor
+   * hasNext
+   * }
+   * dictionaryPhrases: results {
+   * key: field(name: "key") {
+   * value
+   * },
+   * phrase: field(name: "phrase") {
+   * value
+   * }
+   * }
+   * }
+   * }
+   */
+  async fetchDictionaryData(language: string): Promise<DictionaryPhrases> {
+    const cacheKey = this.options.appName + language;
+    const cachedValue = this.getCachedValue(cacheKey);
+    if (cachedValue) {
+      return cachedValue;
+    }
+
+    const dataFetcher = new GraphQLRequestClient(this.options.endpoint);
+    if (!this.options.rootItemId) {
+      this.options.rootItemId = await getAppRoot(dataFetcher, this.options.appName, language);
+    }
+
+    const results = await this.getDictionaryPhrases(dataFetcher, language);
+    return this.cacheValue(cacheKey, results);
+  }
+
+  /**
+   * Gets dictionary phrases
+   * @param {GraphQLRequestClient} dataFetcher
+   * @param {string} language
+   * @returns dictionary phrases
+   */
+  async getDictionaryPhrases(
+    dataFetcher: GraphQLRequestClient,
+    language: string
+  ): Promise<DictionaryPhrases> {
+    const results: DictionaryPhrases = {};
+    let hasNext = true;
+    let after = '';
+
+    while (hasNext) {
+      const fetchResponse = await dataFetcher.request<DictionaryQueryResult>(query, {
+        // `search` query only works with lowercase GUIDs
+        rootItemId: this.options.rootItemId?.toLowerCase(),
+        language,
+        dictionaryEntryTemplateId: SitecoreTemplateId.DictionaryEntry,
+        pageSize: this.options.pageSize,
+        after,
+      });
+
+      fetchResponse.search.dictionaryPhrases.forEach((dictionaryPhrase) => {
+        results[dictionaryPhrase.key.value] = dictionaryPhrase.phrase.value;
+      });
+
+      hasNext = fetchResponse.search.pageInfo.hasNext;
+      after = fetchResponse.search.pageInfo.endCursor;
+    }
+
+    return results;
+  }
+}
+
+// TODO: Move to shared area and reuse for sitemap service (Anastasiya, March 2021)
+const appRootQuery = `
+query GetAppRoot($jssAppTemplateId: String, $appName: String!, $language: String)
+{
+  layout(site: $appName, routePath: "/", language: $language) {
+    homePage: item {
+      appRoot: ancestors(includeTemplateIDs: [$jssAppTemplateId]) {
+        id
+      }
+    }
+  }
+}
+`;
+
+/**
+ * A reply from the GraphQL Sitecore Dictionary Service
+ */
+type AppRootQueryResult = {
```

Comment:
@anastasiya29 Most likely we need to move it to shared area and reuse for sitemap service too? Can you add TODO here?